DMCompiler/Compiler/DM/DMAST.cs

```diff
@@ -1053,6 +1054,20 @@ public class DMASTNewPath : DMASTExpression {
         }
     }
 
+    public class DMASTNewMultidimensionalList : DMASTExpression {
+        public DMASTPath Path;
+        public DMASTCallParameter[] Dimensions;
```

Comment:
This could be `DMASTExpression[]`. There's no need to turn them into parameters.